### PR TITLE
Remove invalid characters from test results

### PR DESCRIPTION
**Pull request by BrennanConroy**
As discussed in https://github.com/dotnet/aspnetcore/pull/26263#issuecomment-698531320

### CODE REVIEW

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs [hunk 1]
```diff
@@ -48,17 +48,30 @@ public Task TestInvalidHeaders(string rawHeaders, string expectedExceptionMessag
...
-        [InlineData("Header: va\0lue", "Malformed request: invalid headers.")]
-        [InlineData("Head\x80r: value", "Invalid characters in header name.")]
-        [InlineData("Header: valu\x80", "Malformed request: invalid headers.")]
-        public Task BadRequestWhenHeaderNameContainsNonASCIIOrNullCharacters(string header, string expectedExceptionMessage)
+        [MemberData(nameof(BadHeaderDataNames))]
+        public Task BadRequestWhenHeaderNameContainsNonASCIIOrNullCharacters(string dataName)
         {
             return TestBadRequest(
-                $"GET / HTTP/1.1\r\n{header}\r\n\r\n",
+                $"GET / HTTP/1.1\r\n{BadHeaderData[dataName].header}\r\n\r\n",
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs:72]: Comment why you're doing this wacky thing 😁

**Review by Tratcher — APPROVED** [review]

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs [hunk 2]
```diff
@@ -48,17 +48,30 @@ public Task TestInvalidHeaders(string rawHeaders, string expectedExceptionMessag
                 expectedExceptionMessage);
         }
 
+        public static Dictionary<string, (string header, string errorMessage)> BadHeaderData => new Dictionary<string, (string, string)>
+        {
+            { "NullInHeaderName", ("Hea\0der: value", "Invalid characters in header name.") },
+            { "NullInHeaderValue", ("Header: va\0lue", "Malformed request: invalid headers.") },
```
Comment (halter73) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs:54]: Does our `HttpParsingData` cause a similar issue? Or is this a problem only with `InlineData` not `MemberData`? If it's the former, we should also fix that. If it's the latter, we can simplify this to a single `IEnumerable<(string, string)>` property for the `MemberData` without changing the actual test.

https://github.com/dotnet/aspnetcore/blob/c39ecb9f9b387ec39112c3eace5ae38db06df160/src/Servers/Kestrel/shared/test/HttpParsingData.cs#L227-L234

Comment (halter73) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs:54]: Nvm. `HttpParsingData` is escaped with our `EscapeNonPrintable` extension method further down in this test class. We should use `EscapeNonPrintable` to generate the key names like we do for `TestInvalidRequestLines` for consistency and to make it more obvious what the test data actually is without looking at the Dictionary.

https://github.com/dotnet/aspnetcore/blob/c39ecb9f9b387ec39112c3eace5ae38db06df160/src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs#L237-L242

Comment (BrennanConroy) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs:54]: It's the \x80 that's causing issues, \0 is fine. So the "MethodWithNullCharData" is safe.

**Review by dougbu — APPROVED** [review]
Is there _also_ an xUnit logger issue here❔

**Review by halter73 — APPROVED** [review]

**Pull request merged**